config.txt
#--------------------------------------------------------------
# astrokit @ 2021  - lGPLv2
# Stephane Gravois - fichier de configuration 
# de astrokit - 
# #--------------------------------------------------------------
# date     | commentaires 
# #--------------------------------------------------------------
# 04/2021     | * ajout configuration des noms
#                 de repertoires et fichiers
#               * suppression des confs inutiles (mis a la fin)
#
# 17/01/2022  | * ajout parametre DEVICE_CLAVIER_USE
#
# 23/01/2022  | * changement type MENU_PAR_DEFAUT int -> char*
#                 pour plus de lisibilite
#               * modification nom CLAVIER en DONNEES puis 
#                 DEVICES_xxx_USE      
#
# avril 2022  | * ajout reductions liees a une poulie en sortie de moteur
#               * CORRECTION reduction moteur d origine des NEQ3 (5760)          
#                 subdivise en 64*90
#               * ajout des reductions R5 R6 et R7 (prevoyant deux poulies)
# juin 2022     * ajout TEMPO_LCD_LOOP et  TEMPO_LCD_DISP
#               * ajout CONFIG_FIC_LED
#               * ajout TEMPO_PID_LOOP
#               * ajout PID_ECH PID_KP PID_KI PID_KD
#               * correction longitude et latitude orgeres
#--------------------------------------------------------------
# Menus possibles par defaut : 
#
#  MENU_AZIMUTAL=0,
#  MENU_EQUATORIAL,
#  MENU_MANUEL_BRUT,
#  MENU_MANUEL_NON_ASSERVI,
#  MENU_MANUEL_ASSERVI,
#--------------------------------------------------------------

MENU_PAR_DEFAUT  MENU_AZIMUTAL
ASTRE_PAR_DEFAUT NGC6756
LED_ETAT_GPIOS   4
TEMPO            50000

# ----------------------- parametres repertoires et fichiers

CONFIG_REP_CAT       cat  
CONFIG_REP_CFG       cfg 
CONFIG_REP_LOG       log   
CONFIG_REP_IN        inp 
CONFIG_REP_OUT       out
CONFIG_REP_SCR       scr
CONFIG_SCR_KERNEL    script_kernel.sh

CONFIG_FIC_LED       /tmp/astrokit-boo-recup-led-ir.txt
CONFIG_FIC_LOG       astrokit.log.new
CONFIG_FIC_DATE      date.txt
CONFIG_FIC_HHMM      hhmm.txt
CONFIG_FIC_PID       pid.csv
CONFIG_FIC_VOUTE     voute.csv

# ----------------------- parametres temporisation
# tempo_menu     doit etre legerement inferieur aux autres tempo
# tempo_raq      doit etre etre deux fois inferieur aux autres tempos
# une valeur non nulle sur TEMPO permet de deduire toutes les autres temporisations par pourcentage
# les valeurs de pourcentage sont dans le code

TEMPO_MENU      240000
TEMPO_RAQ       120000
TEMPO_IR        250010
TEMPO_TERMIOS   250020
TEMPO_CLAVIER   250030 
TEMPO_CAPTEURS  250040
TEMPO_LCD_LOOP  1000000
TEMPO_LCD_DISP  25006
TEMPO_PID_LOOP  5

# ----------------------- parametres regultaur de frequence pid

PID_ECH 1.0
PID_KP  1.0
PID_KI  0.05 
PID_KD  0.05

# ----------------------- parametres acquisitions

DEVICE_USE_INFRAROUGE       1  # flag qui indique utilisation de infrarouge comme source input
DEVICE_USE_KEYBOARD         1  # flag qui indique utilisation du clavier comme source input
DEVICE_USE_LCD              0  # flag qui indique utilisation de ecran LCD1602 / PCA8574
DEVICE_USE_CONTROLER        0  # flag qui indique utilisation des controleurs de moteur externes
DEVICE_USE_CAPTEURS         0  # flag qui indique utilisation de capteurs comme source input
DEVICE_USE_RAQUETTE         0  # flag qui indique utilisation d une raquette (gpio) comme source input
DEVICE_USE_BLUETOOTH        0  # flag qui indique utilisation du bluetooth comme source input

# ----------------------- parametres geographiques

# Latitude et longitude Escout (64)
# LATITUDE   43.28
# LONGITUDE  -0.37
# ALTITUDE   100

# Latitude et longitude Orgeres (35)
# LATITUDE   48.0
# LONGITUDE  -1.67 
# ALTITUDE   100

# Latitude et longitude de test  
# LATITUDE   61.0
# LONGITUDE  -149.9
# ALTITUDE   100

LATITUDE   43.28
LONGITUDE  -0.37
ALTITUDE   100

#---------------------------------------------------------------------------------
# Conf corecte pour carte essai carre (celle qui s enfiche sur rpi3aplus)
# et moteurs en metal NEMA 11 (pas ceux qui sont sur la monture vixen gp)
#--------------------------------------------------------------------------
# ALT_GPIOS    21,26,19,13
# AZI_GPIO     6,5,7,11
# ALT_MASK     0,1,2,3
# AZI_MASK     0,1,2,3
#--------------------------------------------------------------------------
# Masque corect pour carte essai carre (celle qui s enfiche sur rpi3aplus)
#--------------------------------------------------------------------------
# ALT_GPIO     21,26,19,13
# AZI_GPIO     6,5,7,11
# ALT_MASK     3,1,0,2
# AZI_MASK     3,1,0,2
# ----------------------- parametres pwm
# GPIO_FREQUENCE_PWM  250 
#---------------------------------------------------------------------------------

# ----------------------- parametres reductions altitude
# Parametres des moteurs NEQ corriges a la date du 27 mars 2022 : reduction = 5760 
# mesure precisemebnt sur 5 tours en 28800 pas avec le binaire /usr/local/bin/gpio23 

# ALT_GPIO  21,26,19,13  # numeros des gpios pour generation des signaux (4 phases, 2 bobines)
# ALT_MASK  3,1,0,2      # ordre (masque) de generation des signaux (4 phases, 2 bobines)

ALT_GPIO  25,24,23,18  # numeros des gpios pour generation des signaux (4 phases, 2 bobines)
ALT_MASK  0,3,1,2      # ordre (masque) de generation des signaux (4 phases, 2 bobines)
ALT_FPWM  500          # frequence pwm (doit etre au moins superieure a 10 fois la plus grande frequence micropas moteur)
ALT_R1    144      # reduction liee a la monture (nombre de dents de la roue dentee de l axe)
ALT_R2    64       # reducteur du moteur (nombre de pas) = 5760 (64*90) (32*180)
ALT_R3    90       # reducteur du moteur (gearbox)       = 5760 (64*90) (32*180)
ALT_R4    32       # mode micro pas utilise (1/x)
ALT_R5    1        # reduction liee a la poulie
ALT_R6    1        # reduction liee au cpu
ALT_R7    1        # reduction non decrite plus haut ....
ALT_INV   0        # Flag de inversion du sens de rotation
ALT_ACC   2        # Facteur de multiplication en mode MANUEL

# ----------------------- parametres reductions azimut
# Parametres des moteurs NEQ corriges a la date du 27 mars 2022 : reduction = 5760 
# mesure precisemebnt sur 5 tours en 28800 pas avec le binaire /usr/local/bin/gpio23 

# AZI_GPIO  6,5,7,11     # numeros des gpios pour generation des signaux (4 phases, 2 bobines)
# AZI_MASK  3,1,0,2      # ordre (masque) de generation des signaux (4 phases, 2 bobines)

AZI_GPIO  26,19,13,6     # numeros des gpios pour generation des signaux (4 phases, 2 bobines)
AZI_MASK  0,3,1,2       # ordre (masque) de generation des signaux (4 phases, 2 bobines)
AZI_FPWM  500          # frequence pwm (doit etre au moins superieure a 10 fois la plus grande frequence micropas moteur)
AZI_R1    144      # reduction lee a la monture (nombre de dents de la roue dentee de l axe)
AZI_R2    64       # reducteur du moteur (nombre de pas) : R total = 5760 (64*90) (32*180)
AZI_R3    90       # reducteur du moteur (gearbox) : R total = 5760 (64*90) (32*180)
AZI_R4    32       # mode micro pas utilise (1/x)
AZI_R5    1        # reduction liee a la poulie
AZI_R6    1        # reduction liee au cpu
AZI_R7    1        # reduction non decrite plus haut ....
AZI_INV   1        # Flag de inversion du sens de rotation
AZI_ACC   2        # Facteur de multiplication en mode MANUEL

# ----------------------- parametres reductions altaz

ALTAZ_FORWARD        1.005   # affinage vitesses : acceleration globale tres lente (azi + alt)
ALTAZ_REWIND         1.005   # affinage vitesses : acceleration globale tres lente (azi + alt)

ALTAZ_FORWARD_FAST   1.5     # acceleration globale rapide (azi + alt)
ALTAZ_REWIND_FAST    1.5     # acceleration globale rapide (azi + alt)

#===============================================================
# anciens parametres qui ne servent plus (notamment avec 
#  utilisation de la raquette infrarouge )
#===============================================================

#--------------------------------------------------------------
#  numero des ports GPIO
#--------------------------------------------------------------

GPIO_DIR_ALT 21    # direction de l'altitude
GPIO_CLK_ALT 20    # horloge de l'altitude
GPIO_SLP_ALT 16    # sleep du controleur du moteur
GPIO_RST_ALT 12    # reset du controleur du moteur
GPIO_MMM_ALT 7     # mode micro pas du controleur du moteur
GPIO_ENA_ALT 8     # broche enable du controleur du moteur

GPIO_DIR_AZI 25    # direction de l'azimut
GPIO_CLK_AZI 24    # horloge de l'azimut
GPIO_SLP_AZI 23    # sleep du controleur du moteur
GPIO_RST_AZI 18    # reset du controleur du moteur
GPIO_MMM_AZI 15    # mode micro pas du controleur du moteur
GPIO_ENA_AZI 14    # broche enable du controleur du moteur

#--------------------------------------------------------------
# parametres de RAQUETTE [ GPIO doivent etre en INPUT ] 
#--------------------------------------------------------------

GPIO_RAQ_O   19    # raquette direction OUEST
GPIO_RAQ_E   6     # raquette direction EST
GPIO_RAQ_S   13    # raquette direction SUD
GPIO_RAQ_N   26    # raquette direction NORD
GPIO_RAQ_V   5     # raquette VALIDER

#--------------------------------------------------------------
# parametres de CLAVIER 4 * 4 = 16 touches (16 switchs)
# [ LIGNES   doivent etre en INPUT  ]
# [ COLONNES doivent etre en OUTPUT ]
#--------------------------------------------------------------

GPIO_KEY_L1  10   # entree keyboard ligne 1
GPIO_KEY_L2  9    # entree keyboard ligne 2
GPIO_KEY_L3  11   # entree keyboard ligne 3
GPIO_KEY_L4  5    # entree keyboard ligne 4
GPIO_KEY_C1  6    # sortie keyboard colonnne 1
GPIO_KEY_C2  13   # sortie keyboard colonnne 2
GPIO_KEY_C3  19   # sortie keyboard colonnne 3
GPIO_KEY_C4  26   # sortie keyboard colonnne 4

# ----------------------- parametres gpios

# la led est parametree grace a :
# insmod lirc_rpi.ko gpio_in_pin=27 dans /etc/init.d/rcS
# GPIO_LED_IR     27  
# inutiles (2021) :
# GPIO_INPUT      7,8 
# GPIO_OUTPUT     14,15 

#===============================================================
# fin fichier - fin fichier - fin fichier - fin fichier - fin fi
#===============================================================





cfg/config.txt
#--------------------------------------------------------------
# astrokit @ 2021  - lGPLv2
# Stephane Gravois - fichier de configuration 
# de astrokit - 
# #--------------------------------------------------------------
# date     | commentaires 
# #--------------------------------------------------------------
# 04/2021  | ajout configuration des noms
#            de repertoires et fichiers
#            suppression des confs inutiles (mis a la fin)
#--------------------------------------------------------------

MENU_PAR_DEFAUT  0 
ASTRE_PAR_DEFAUT PLA0

# ----------------------- parametres repertoires et fichiers

CONFIG_REP_CAT       cat
CONFIG_REP_CFG       cfg
CONFIG_REP_LOG       log
CONFIG_REP_IN        inp
CONFIG_REP_OUT       out
CONFIG_FIC_LOG       astrokit.log
CONFIG_FIC_DATE      date.txt
CONFIG_FIC_HHMM      hhmm.txt

# ----------------------- parametres temporisation

TEMPO_MENU      50000
TEMPO_RAQ       7000
TEMPO_IR        70000 
TEMPO_CLAVIER   52000
TEMPO_CAPTEURS  52000

# ----------------------- parametres acquisitions

DONNEES_CONTROLEUR   0  # flag qui indique utilisation des controleurs de moteur externes
DONNEES_CAPTEURS     0  # flag qui indique utilisation de capteurs 
DONNEES_RAQUETTE     0  # flag qui indique utilisation d une raquette 
DONNEES_BLUETOOTH    0  # flag qui indique utilisation du bluetooth 
DONNEES_INFRAROUGE   1  # flag qui indique utilisation de infrarouge

# ----------------------- parametres geographiques

LATITUDE   43.3
LONGITUDE -0.39 
ALTITUDE   200

# ----------------------- parametres gpios
# la led est parametree grace a :
# insmod lirc_rpi.ko gpio_in_pin=27 dans /etc/init.d/rcS
# GPIO_LED_IR     27  
# inutiles (2021) :
# GPIO_INPUT      7,8 
# GPIO_OUTPUT     14,15 

GPIO_LED_ETAT   4 
GPIO_ALT        21,26,19,13
GPIO_AZI        6,5,7,11
GPIO_MASQUE     3,1,0,2

# ----------------------- parametres pwm

GPIO_FREQUENCE_PWM  1500 

# ----------------------- parametres reductions altitude

ALT_R1  144      # reduction liee a la monture ( = 65 dents / 1.5 = 43.333)
ALT_R2  266      # reducteur du moteur = 28.75
ALT_R3  22       # nombre de pas du moteur en altitude
ALT_R4  51       # mode micro pas utilise (1/x)
ALT_ROT 0        # Flag de reversibilite du sens de rotation (en cas d'erreur) val = 0 ou 1
ALT_ACC 2        # Facteur de multiplication en mode MANUEL

# ----------------------- parametres reductions azimut

AZI_R1   144     # reduction lee a la monture ( = 130 dents)
AZI_R2   266     # reducteur du moteur = 28.75
AZI_R3   22      # nombre de pas du moteur en azimut
AZI_R4   51      # mode micro pas utilise (1/x)
AZI_ROT  0       # Flag de reversibilite du sens de rotation (en cas d'erreur) val = 0 ou 1
AZI_ACC  2       # Facteur de multiplication en mode MANUEL

# ----------------------- parametres reductions altaz

ALTAZ_FORWARD        1.01   # acceleration globale        (azi + alt)
ALTAZ_REWIND         1.01   # acceleration globale        (azi + alt)
ALTAZ_FORWARD_FAST   1.5    # acceleration globale rapide (azi + alt)
ALTAZ_REWIND_FAST    1.5    # acceleration globale rapide (azi + alt)

#===============================================================
# anciens parametres qui ne servent plus (notamment avec 
#  utilisation de la raquette infrarouge )
#===============================================================

#--------------------------------------------------------------
#  numero des ports GPIO
#--------------------------------------------------------------

GPIO_DIR_ALT 21    # direction de l'altitude
GPIO_CLK_ALT 20    # horloge de l'altitude
GPIO_SLP_ALT 16    # sleep du controleur du moteur
GPIO_RST_ALT 12    # reset du controleur du moteur
GPIO_MMM_ALT 7     # mode micro pas du controleur du moteur
GPIO_ENA_ALT 8     # broche enable du controleur du moteur

GPIO_DIR_AZI 25    # direction de l'azimut
GPIO_CLK_AZI 24    # horloge de l'azimut
GPIO_SLP_AZI 23    # sleep du controleur du moteur
GPIO_RST_AZI 18    # reset du controleur du moteur
GPIO_MMM_AZI 15    # mode micro pas du controleur du moteur
GPIO_ENA_AZI 14    # broche enable du controleur du moteur

#--------------------------------------------------------------
# parametres de RAQUETTE [ GPIO doivent etre en INPUT ] 
#--------------------------------------------------------------

GPIO_RAQ_O   19    # raquette direction OUEST
GPIO_RAQ_E   6     # raquette direction EST
GPIO_RAQ_S   13    # raquette direction SUD
GPIO_RAQ_N   26    # raquette direction NORD
GPIO_RAQ_V   5     # raquette VALIDER

#--------------------------------------------------------------
# parametres de CLAVIER 4 * 4 = 16 touches (16 switchs)
# [ LIGNES   doivent etre en INPUT  ]
# [ COLONNES doivent etre en OUTPUT ]
#--------------------------------------------------------------

GPIO_KEY_L1  10   # entree keyboard ligne 1
GPIO_KEY_L2  9    # entree keyboard ligne 2
GPIO_KEY_L3  11   # entree keyboard ligne 3
GPIO_KEY_L4  5    # entree keyboard ligne 4

GPIO_KEY_C1  6    # sortie keyboard colonnne 1
GPIO_KEY_C2  13   # sortie keyboard colonnne 2
GPIO_KEY_C3  19   # sortie keyboard colonnne 3
GPIO_KEY_C4  26   # sortie keyboard colonnne 4

#===============================================================
# fin fichier - fin fichier - fin fichier - fin fichier - fin fi
#===============================================================





config.c

MENU_PAR_DEFAUT  MENU_AZIMUTAL
ASTRE_PAR_DEFAUT NGC6755

//  ----------------------- parametres repertoires et fichiers 

CONFIG_REP_CAT       cat
CONFIG_REP_CFG       cfg
CONFIG_REP_LOG       log
CONFIG_REP_IN        inp
CONFIG_REP_OUT       out
CONFIG_REP_SCR       scr
CONFIG_FIC_LED       /tmp/astrokit-boo-recup-led-ir.txt
CONFIG_FIC_LOG       astrokit.log.new
CONFIG_FIC_DATE      date.txt
CONFIG_FIC_HHMM      hhmm.txt

CONFIG_SCR_KERNEL   script_kernel.sh

//  ----------------------- parametres temporisation                       
//  tempo_menu     doit etre legerement inferieur aux autres tempo         
//  tempo_raq      doit etre etre deux fois inferieur aux autres tempos    

TEMPO_MENU      1000000
TEMPO_RAQ       500000
TEMPO_IR        1000010
TEMPO_TERMIOS   1000020
TEMPO_CLAVIER   50000
TEMPO_CAPTEURS  50000
TEMPO_LCD_LOOP  1000000
TEMPO_LCD_DISP  1000000
TEMPO_PID_LOOP  5       

//  ----------------------- parametres asservissement de frequence PID 

PID_ECH 1.0
PID_KP  1.0
PID_KI  0.05 
PID_KD  0.05

//  ----------------------- parametres acquisitions 

DEVICE_USE_INFRAROUGE       1  //  flag qui indique utilisation de infrarouge comme source input
DEVICE_USE_KEYBOARD         1  //  flag qui indique utilisation du clavier comme source input
DEVICE_USE_LCD              0  //  flag qui indique utilisation de ecran LCD1602 / PCA8574
DEVICE_USE_CONTROLER        0  //  flag qui indique utilisation des controleurs de moteur externes
DEVICE_USE_CAPTEURS         0  //  flag qui indique utilisation de capteurs comme source input
DEVICE_USE_RAQUETTE         0  //  flag qui indique utilisation d une raquette (gpio) comme source input
DEVICE_USE_BLUETOOTH        0  //  flag qui indique utilisation du bluetooth comme source input

//  ----------------------- parametres geographiques 

//  Latitude et longitude Escout (64) 

LATITUDE   43.28
LONGITUDE  -0.37
ALTITUDE   100

//  Latitude et longitude Orgeres (35) 
//  LATITUDE   48.0 
//  LONGITUDE  -1.67 
//  ALTITUDE   100 

//  Latitude et longitude de test   
//  LATITUDE   61.0 
//  LONGITUDE  -149.9 
//  ALTITUDE   100 

// -------------------------------------------------------------------------- 
//  Conf corecte pour carte essai carre (celle qui s enfiche sur rpi3aplus) 
//  et moteurs en metal NEMA 11 (pas ceux qui sont sur la monture vixen gp) 
// -------------------------------------------------------------------------- 

GPIO_LED_ETAT   4
GPIO_ALT        21,26,19,13
GPIO_AZI        6,5,7,11
//  GPIO_MASQUE     0,1,2,3 

// -------------------------------------------------------------------------- 
//  Masque corect pour carte essai carre (celle qui s enfiche sur rpi3aplus) 
// -------------------------------------------------------------------------- 

//  GPIO_LED_ETAT   4 
//  GPIO_ALT        21,26,19,13 
//  GPIO_AZI        6,5,7,11 

GPIO_MASQUE     3,1,0,2

//  ----------------------- parametres pwm 

GPIO_FREQUENCE_PWM  250 

//  ----------------------- parametres reductions altitude 
//  Parametres des moteurs NEQ corriges a la date du 27 mars 2022 : reduction = 5760  
//  mesure precisemebnt sur 5 tours en 28800 pas avec le binaire /usr/local/bin/gpio23  

ALT_R1  144      //  reduction liee a la monture (nombre de dents de la roue dentee de l axe) 
ALT_R2  100       //  reducteur du moteur (nombre de pas) : R total = 5760 (64*90) (32*180) 
ALT_R3  90       //  reducteur du moteur (gearbox) : R total = 5760 (64*90) (32*180) 
ALT_R4  8       //  mode micro pas utilise (1/x) 
ALT_R5  1        //  reduction liee a la poulie 
ALT_R6  1        //  reduction liee au cpu 
ALT_R7  1        //  reduction non decrite plus haut .... 
ALT_ROT 0        //  Flag de reversibilite du sens de rotation 
ALT_ACC 2        //  Facteur de multiplication en mode MANUEL 

//  ----------------------- parametres reductions azimut 

AZI_R1  144      //  reduction lee a la monture (nombre de dents de la roue dentee de l axe) 
AZI_R2  100       //  reducteur du moteur (nombre de pas) : R total = 5760 (64*90) (32*180) 
AZI_R3  90       //  reducteur du moteur (gearbox) : R total = 5760 (64*90) (32*180) 
AZI_R4  8       //  mode micro pas utilise (1/x) 
AZI_R5  1        //  reduction liee a la poulie 
AZI_R6  1        //  reduction liee au cpu 
AZI_R7  1        //  reduction non decrite plus haut .... 
AZI_ROT 1        //  Flag de reversibilite du sens de rotation 
AZI_ACC 2        //  Facteur de multiplication en mode MANUEL 

//  ----------------------- parametres reductions altaz 

ALTAZ_FORWARD        1.005   //  affinage vitesses : acceleration globale tres lente (azi + alt) 
ALTAZ_REWIND         1.005   //  affinage vitesses : acceleration globale tres lente (azi + alt) 

ALTAZ_FORWARD_FAST   1.5     //  acceleration globale rapide (azi + alt) 
ALTAZ_REWIND_FAST    1.5     //  acceleration globale rapide (azi + alt) 



